### PR TITLE
[FEATURE] Doktype lookup

### PR DESCRIPTION
**Pull request by saitho**
A new setting "withDoktype" is now evaluated
and will find all pages with the matching doktype.

Resolves: #14

### CODE REVIEW

**Comment by smichaelsen** [comment]
please include only the changes in the MR, that are required for your feature. no changes to gitignore, composer.json general whitespacing etc

File: Classes/ConfigurationModifier.php [hunk 1]
```diff
@@ -43,6 +43,12 @@ protected static function findDynamicPages(array $dynamicPagesConfiguration): ar
             self::$cache[$withPluginsCacheKey] = self::$cache[$withPluginsCacheKey] ?? self::findPagesWithPlugins($withPlugins);
             array_push($pageUids, ...self::$cache[$withPluginsCacheKey]);
         }
+        if (isset($dynamicPagesConfiguration['withDoktype'])) {
+            $withDoktype = is_array($dynamicPagesConfiguration['withDoktype']) ? $dynamicPagesConfiguration['withDoktype'] : [$dynamicPagesConfiguration['withDoktype']];
```
Comment (smichaelsen) [inline Classes/ConfigurationModifier.php:47]: Please rename the configuration option to `withDoktypes` (plural).

The doktypes should be integers, but there's no sanitaziation in place. You pass it to the QueryBuilder as `PARAM_INT_ARRAY` - meaning it's not escaped. I'm not sure this is safe.

I'd prefer to have sanitization like `$withDoktypes = array_map('intval', $withDoktypes);`.

Comment (saitho) [reply Classes/ConfigurationModifier.php]: Done

**Comment by smichaelsen** [comment]
Thank you @saitho

**Pull request merged**